data/models/users-model.js

```diff
@@ -0,0 +1,23 @@
+const db = require('../dbConfig');
+
+const checkUsername = async username => {
+  // Checks to see if username is taken. Since I don't understand security, I thought it make be good to return 'taken' or 'available' rather than risk returning the hashed password.
```

Comment:
good idea for a check--you never want to tell a user "that password is already taken" but since you want each username to be unique, this is a good habit